Use publishing api to retrieve content items

theleebriggs opened the pull request:
### Motivation
The `content_items` import part of the application is now the only part still using a combination of `search-api` and `content-store`. 

It is desirable for our import services to be as unified in their approach as possible, in addition to this the publishing api contains a richer set of information e.g. revision history.
https://trello.com/c/VBqWEPj8/188-8-use-publishing-api-to-retrieve-the-content-items

### Description
Replaces calls to `content_store` and `search_api`. Instead content is retrieved by `document_type` from the `publishing-api`.

The document types are configured in a .yml file for now, this will change once there is an endpoint on the `publishing-api` to retrieve a live list.

When importing `content-items` the associated organisations and taxonomies are retrieved using the publishing api's `expanded_links` endpoint.

Calls to get/import content by organisation have been removed along with their respective code.

pmanrubia commented on app/models/importers/all_content_items.rb:
```diff
@@ -1,10 +1,20 @@
+
     def run
-      content_item_importer = ContentItemsByOrganisation.new
+      document_types = Rails.configuration.document_types
```
I would create `attr_writer` with the document_types so it can be updated in the test, and I would create a reader this way:

```ruby
def document_types
  @document_types ||= Rails.configuration.document_types
```

theleebriggs replied:
Implemented as you suggested now.

pmanrubia commented on app/services/content_items_service.rb:
```diff
@@ -1,23 +1,27 @@
-        content_item[:taxons] = TaxonomyParser.parse(content_item)
-        yield content_item
-      end
+    publishing_api.find_each(query_fields, build_query_options(document_type)) do |content_item|
```
what about extracting a variable:

```ruby
query_options = { document_type: document_type, expand_links: true }
```

theleebriggs replied:
As discussed, keeping this as a function for now.

theleebriggs commented on app/services/clients/publishing_api.rb:
```diff
@@ -15,19 +15,28 @@ def initialize
+          if options[:expand_links]
+            result[:expanded_links] = expand_links(result["content_id"])
+          end
+          yield result.symbolize_keys
```
Updated now.

theleebriggs commented on spec/services/clients/publishing_api_spec.rb:
```diff
@@ -3,39 +3,62 @@
-    subject.publishing_api = double
-    page_1 = { "results" => [{}], "pages" => 2, "current_page" => 1 }
-    page_2 = { "results" => [{}], "pages" => 2, "current_page" => 2 }
+  describe "#find_each" do
```
Agreed, will keep this in mind for next PR.

theleebriggs commented on spec/services/clients/publishing_api_spec.rb:
```diff
@@ -3,39 +3,62 @@
+    end
+
+    it "yields the results" do
+      yielded = []
```
Updated now.

theleebriggs commented on spec/services/clients/publishing_api_spec.rb:
```diff
@@ -3,39 +3,62 @@
-      fields: [:field_1, :field_2]
-    }
-    result = { "results" => [{}], "pages" => 1, "current_page" => 1 }
+  describe "#expand_links" do
```
Is now a private method.

theleebriggs commented:
Thanks for the review @pmanrubia - I've addressed your comments now - both the code changes and adding more details to the commit notes on getting `links` from the publishing api.

I've also switched from using `expanded_links` to just `links` to speed up the import. This means there are 2 additional commits to add `content_id` to the org table (1608429551a51bbccf1a1188b81a4c59c040806b and 6cf5a09b541f11a5d2d65128a9b854b2d91e8b2d) - this enables us to associate orgs with content items by content_id - the only field returned by the `/links/CONTENT_ID` endpoint.

I have some speed test figures to run past you as well - maybe before this is merged?

Cheers.

pmanrubia commented:
LGTM 👍
